Library/Homebrew/formula_cellar_checks.rb

```diff
@@ -275,6 +275,47 @@ def check_python_symlinks(name, keg_only)
     "Python formulae that are keg-only should not create `pip3` and `wheel3` symlinks."
   end
 
+  INVALID_EMPTY_FILE_NAMES = %w[AUTHORS ChangeLog changes COPYING NEWS README TODO].freeze
+  INVALID_EMPTY_FILE_EXTENSIONS = %w[.html .md .rst .txt].freeze
+  INVALID_EMPTY_FILE_LOCATIONS = %w[bin libexec share/man share/doc].freeze
+  # Names and extensions to ignore, even if they are in a location within prefix given above
+  VALID_EMPTY_FILE_NAMES = %w[__init__.py REQUESTED].freeze
+  VALID_EMPTY_FILE_EXTENSIONS = %w[.stamp].freeze
+
+  def check_empty_files(prefix)
+    return unless prefix.directory?
+
+    empty_files = []
+    prefix.children.each do |f|
+      next if INVALID_EMPTY_FILE_NAMES.none?(File.basename(f)) &&
+              INVALID_EMPTY_FILE_EXTENSIONS.none?(File.extname(f))
+      next unless File.zero?(f)
+
+      empty_files << f
+    end
+
+    INVALID_EMPTY_FILE_LOCATIONS.each do |loc|
+      next unless (prefix/loc).directory?
+
+      # search all directories in INVALID_EMPTY_FILE_LOCATIONS and their subdirectories
+      Dir.glob(prefix/loc/"**/*").each do |f|
```

Comment:
```suggestion
      Pathname.glob(path/"**/*").each do |p|
```

are you sure this finds `.foo` files?